app/src/main/java/com/ekeitho/reduxy/MainActivity.kt

```diff
@@ -48,22 +49,20 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
 
             // subjects are observers and observables
             // therefore turning it just to an observable - means that VM CAN'T MAKE ANY CHANGES TO THE STATE!!!
-            ViewModel.init(ApplicationStateStream.observableState.hide())
+
+            val stateStream = BehaviorSubject.createDefault(ApplicationState("Keith", 25, true))
+            val eventStream : Subject<Event> = PublishSubject.create()
```

Comment:
We don't have to worry anymore about classes accessing these when they shouldn't, and since the VM and Reducer are never destroyed, it won't have to be injected more than once.